ts/src/binance.ts

```diff
@@ -6921,6 +6921,14 @@ export default class binance extends Exchange {
         if (side !== 'long') {
             quantity = Precise.stringMul ('-1', quantity);
         }
+        const optionSide = this.safeString (position, 'optionSide');
+        let optionType = undefined;
+        if (optionSide === 'CALL') {
+            optionType = 'call';
+        } else if (optionSide === 'PUT') {
+            optionType = 'put';
+        }
```

Comment:
we could use `safeStringLower` here instead 🙂

Reply:
I intentionally avoided that, because wanted it to be explicit, but yeah, can be used.

Reply:
I also don't know (havent checked the docs) if they might return other value there (other than `put/call`)

Reply:
@ttodua I don't think so, even if they have it's better to provide the exchange-specific side than none

Reply:
@carlosmiei sorry, I was late to commit the change, as I see you've already done it. this PR is ready, not?

Reply:
@carlosmiei lmk if we miss anything

Reply:
@carlosmiei this PR fixes a possible conflict with upcoming fetchPositionFull, so if you can take a 10-second look into this and approve this PR, would be good. thanks

Reply:
ah, i've not evaluated Dan's comment correctly..
well, if you think we should not add separate `parseOptionPosition`, then lmk and I'll close this PR.
I assumed that options market's position might need different values to be added in mandatory structure, but @lead should decide.

Reply:
@Dan-krm can you proactively determine, whether options markets need to have different mandatory fields in response?
1) what is current situation on different exchanges - can we see any similarities and guaranteed provided fields in top exchanges, so we could decisively say to include some additional fields in options' position fields? 
2) are those fields important for users?

Reply:
The only additional fields would probably be optionType, strike and expiry which might be useful as separate fields apart from the symbol, I don't think any additional fields are really required. Using a separate parser doesn't seem like a necessity to me at the moment but I'm not opposed to it and it might be needed at some point

In the future if crypto options markets mature it might be useful to have an option style field (american, european, asian, binary etc.) The greeks and implied volatility of the different options would be useful but I think we should create new unified methods for those things. Also a field to tell if the position is a contract for `writing` or `purchasing` an option would be good to have but I'm not seeing these fields returned by any exchange fetch option position endpoints currently

I think having a separate parser for options positions makes sense for making the type, strike and expiry more accessible and for future proofing so we can easily add more fields if the exchange API's have more data available in the future

Reply:
thanks @Dan-krm .
@carlosmiei can you confirm?

Reply:
@carlosmiei

Reply:
Having different parsers is totally ok for me we do it extensively already when the raw response is very different depending on the market type (eg `parseSpotTrade` and `parseContractTrade`)

Reply:
I'm just not sure if we should have "options only" values or not, for instance our market structure has the field `strike` `expiry` etc etc even if the market is spot

Reply:
@Dan-krm can you confirm the following:
A) in 'market' we do have all needed info  - option type, expiry, strike price, etc.. lets name this `"X data"`
B) however, when doing parsing, if market is somewhat unrecognized (i.e. newly added or retrieved from websockets etc, where for some reason we can't derive the 'market' object), then user will loose the info of that `"X data"`. so, that's why I was thinking that the essential fields were included in 'response' structure too, so we didn't rely on market object identification.

@Dan-krm @carlosmiei wdyt?

Reply:
The points Carlos made seem valid, since the type, strike and expiry will always be attached to a specific option contract that can be fetched through the symbol or the market structure maybe we don't need a separate parser to add those values at this time

Reply:
Dan, but did you read my (B) point? the thing is if we get the position from non-contract-specific calls (i.e. bulk fetchPositions, websockets, etc..) where symbol is not recognized (happens sometimes), then those data will be absent and neither in structure we would have.

Reply:
anyway, as you decide, lmk @carlosmiei , should I close this PR?